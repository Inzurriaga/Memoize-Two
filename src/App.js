import React, { Component } from 'react';
import Login from "./Login"
import Header from "./Header"
import MainContent from "./MainContent"

class App extends Component {
  constructor(){
    super();
    this.state = {
      users: [],
      currentuser: undefined
    }
  }

  componentDidMount() {
    let user = JSON.parse(localStorage.getItem("player"))
    if(user !== null){
      console.log("there is players")
      this.setState({
        users: user
      })
    }else{
      console.log("there is no players")
    }
  }

  createNewAccount = (player) => {
    this.state.users.push(player);
    localStorage.setItem("player", JSON.stringify(this.state.users))
    this.setState({
      currentuser: player
    })
  }

  loginToAccount = (player) => {
    this.setState({
      currentuser: player
    })
  }


  render() {
    return (
      <div className="App">
        <Header />
        { !this.state.currentuser ? 
          <Login createNewAccount={this.createNewAccount} users={this.state.users} loginToAccount={this.loginToAccount}/> :
          <MainContent />
        }
      </div>
    );
  }
}

export default App;
