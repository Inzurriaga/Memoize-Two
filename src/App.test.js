import React from 'react';
import { shallow } from 'enzyme';
import App from "./App"

describe("App", ()=> {
  let wrapper; 
  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
  })

  it("should match Snap shot", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should have a default state", () => {
    expect(wrapper.state()).toEqual({
      users: [],
      currentuser: undefined,
      page: "home",
      Questiontoggle: false,
      selectedQuestion: {}
    })
  })

  it("should add user to array and set currentuser when createaccount is invoke", () => {
    expect(wrapper.state("currentuser", "users")).toEqual(undefined, [])
    wrapper.instance().createNewAccount(
      {"name":"gabriel","password":"gabriel","completed":[1,2,5]})
      expect(wrapper.state("currentuser", "users")).toEqual({"name":"gabriel","password":"gabriel","completed":[1,2,5]}, [{"name":"gabriel","password":"gabriel","completed":[1,2,5]}])
  })

  it("should add user to currentuser when login is invoke", () => {
    expect(wrapper.state("currentuser")).toEqual(undefined)
    wrapper.instance().loginToAccount({"name":"gabriel","password":"gabriel","completed":[1,2,5]})
    expect(wrapper.state("currentuser")).toEqual({"name":"gabriel","password":"gabriel","completed":[1,2,5]})
  })

  it("should add user to currentuser when login is invoke", () => {
    expect(wrapper.state("currentuser", "page")).toEqual(undefined, "home")
    wrapper.instance().signOut()
    expect(wrapper.state("currentuser", "page")).toEqual(undefined, "home")
  })

  it("should toggle page when toggle page is invoke", () => {
    expect(wrapper.state("Questiontoggle", "page")).toEqual(false, "home")
    wrapper.instance().togglePage("train")
    expect(wrapper.state("Questiontoggle", "page")).toEqual(false, "train")
  })

  it("should toggle to the question when toggle question is invoke", () => {
    expect(wrapper.state("Questiontoggle", "selectedQuestion")).toEqual(false, {})
    wrapper.instance().toggleToQuestion({title: "number return 5", type: "number", instructions: "return the number 5", answer: 5, startup: "function number() {\n//code here \n} \n\n number()", id: 1
  })
    expect(wrapper.state("Questiontoggle", "page")).toEqual(true, {title: "number return 5", type: "number", instructions: "return the number 5", answer: 5, startup: "function number() {\n//code here \n} \n\n number()", id: 1
  })
  })

})