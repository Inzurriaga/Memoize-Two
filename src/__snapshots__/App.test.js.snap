// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match Snap shot 1`] = `
<div
  className="App"
>
  <Header
    signOut={[Function]}
    togglePage={[Function]}
  />
  <login
    createNewAccount={[Function]}
    loginToAccount={[Function]}
    users={Array []}
  />
</div>
`;
